### PR TITLE
[SPARK-43649][SPARK-43650][SPARK-43651][SQL] Assign names to the error class _LEGACY_ERROR_TEMP_240[1-3]

### PR DESCRIPTION
**Pull request by beliefer**
### What changes were proposed in this pull request?
The pr aims to assign a name to the error class _LEGACY_ERROR_TEMP_240[1-3].

### Why are the changes needed?
Improve the error framework.

### Does this PR introduce _any_ user-facing change?
'No'.

### How was this patch tested?
Exists test cases.

### CODE REVIEW

**Review by MaxGekk — CHANGES_REQUESTED** [review]
Could you add the error class `INVALID_LIMIT_LIKE` with error format like `The limit like clause <expr> is invalid.`, and add sub-classes:
- `DATA_TYPE`: `Expects ..., but got ... `
- `IS_NEGATIVE`
- `IS_NULL`

**Comment by beliefer** [comment]
> 

Got it.

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/CheckAnalysis.scala [hunk 1]
```diff
@@ -85,26 +85,28 @@ trait CheckAnalysis extends PredicateHelper with LookupCatalog with QueryErrorsB
...
           "name" -> name,
-          "limitExpr" -> toSQLExpr(limitExpr)))
+          "expr" -> toSQLExpr(limitExpr)))
       case e if e.dataType != IntegerType => limitExpr.failAnalysis(
-        errorClass = "_LEGACY_ERROR_TEMP_2401",
+        errorClass = "INVALID_LIMIT_LIKE_EXPRESSION.DATA_TYPE",
         messageParameters = Map(
           "name" -> name,
+          "expr" -> toSQLExpr(limitExpr),
           "dataType" -> e.dataType.catalogString))
```
Comment (MaxGekk) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/CheckAnalysis.scala:97]: Please, wrap it by `toSQLType()`

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/CheckAnalysis.scala [hunk 2]
```diff
@@ -85,26 +85,28 @@ trait CheckAnalysis extends PredicateHelper with LookupCatalog with QueryErrorsB
...
               "name" -> name,
-              "limitExpr" -> limitExpr.sql))
+              "expr" -> toSQLExpr(limitExpr)))
           case v: Int if v < 0 => limitExpr.failAnalysis(
-            errorClass = "_LEGACY_ERROR_TEMP_2403",
+            errorClass = "INVALID_LIMIT_LIKE_EXPRESSION.IS_NEGATIVE",
             messageParameters = Map(
               "name" -> name,
+              "expr" -> toSQLExpr(limitExpr),
               "v" -> v.toString))
```
Comment (MaxGekk) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/CheckAnalysis.scala:110]: Wrap it by `toSQLValue(v, IntegerType)`

File: sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/analysis/AnalysisErrorSuite.scala [hunk 3]
```diff
@@ -653,7 +654,8 @@ class AnalysisErrorSuite extends AnalysisTest {
   errorTest(
```
Comment (MaxGekk) [inline sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/analysis/AnalysisErrorSuite.scala:654]: The same, please, use `errorClassTest`.

**Review by MaxGekk — CHANGES_REQUESTED** [review]

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/CheckAnalysis.scala (hunk 1 above)
Comment (beliefer) [reply sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/CheckAnalysis.scala:97]: Got it.

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/CheckAnalysis.scala (hunk 2 above)
Comment (beliefer) [reply sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/CheckAnalysis.scala:110]: Got it.

File: sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/analysis/AnalysisErrorSuite.scala [hunk 4]
```diff
@@ -629,7 +629,8 @@ class AnalysisErrorSuite extends AnalysisTest {
...
```
[the remark this answers is not in the document]
Comment (beliefer) [reply sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/analysis/AnalysisErrorSuite.scala:629]: I see.

**Review by MaxGekk — APPROVED** [review]

**Comment by MaxGekk** [comment]
+1, LGTM. Merging to master.
Thank you, @beliefer.

**Comment by beliefer** [comment]
@MaxGekk Thank you!

**Pull request closed without merging**